### ISSUE TITLE
Support for water leak sensor 

### ISSUE THREAD

**Issue by jsrivatsan** [dgreif/ring#195]
Can you please add support for water leak sensor that is part of the ring security system? This plugin is awesome and has made home automation a breeze. Keep up the great work!

**Comment by dgreif** [comment]
@jsrivatsan thanks for the request.  I should be able to copy what I have in place for the smoke/CO sensor and get pretty close.  I'll need to have you test it out for me since I don't have one of these sensor, but it should be a lot easier than testing the smoke/CO sensor 😉.  No promises on time-line, but it should be pretty straightforward.

One small issue I can see already is that there is no "Freeze" sensor in HomeKit.  They have smoke, CO, and leak (which I will use for "Flood"), but no Freeze.  I could use a "Contact", "Occupancy" or "Smoke" as a substitute for "Freeze".   It would work just fine for automations with all three of those, but the icon won't be indicative of "Freeze".  Any preference?

**Comment by jsrivatsan** [comment]
Thanks a ton. I will be more than happy to test it out. I am primarily adding the sensor to detect water leak. So I can definitely test that out thoroughly. For the freeze, I do not have any particular preference. May be we can use use a “contact”?

**Comment by dgreif** [comment]
@jsrivatsan I've got a release for you to test out 🎁.  `npm i -g `homebridge-ring@5.11.0-alpha.0` should install it for you assuming you do global installs for your homebridge packages.  Please note this is an `alpha` release so it won't be installable through `homebridge-config-ui-x`.  Once installed, restart your homebridge server and you should see the flood/freeze sensor show up.  There is a "Flood Sensor" and "Freeze Sensor" within the same HomeKit device, and they should each work for their respective job.  If you could please test out both sensors and let me know the results, then I can get this out in a real release.

**Comment by jsrivatsan** [comment]
Thanks for adding this. I guess I have something to play with over the weekend :-)

**Comment by jsrivatsan** [comment]
I updated the plugin as you suggested but when I removed and added the ring hub again in my home app, I got a warning saying I need to setup the accessories first in the manufacturer's app. None of the accessories showed up in the home app. 

I am using multiple homebridge instances as it is easier to manage each plugin. I setup multiple instances based on the instructions here (https://gist.github.com/swiss6th/dfb74278d8318d4e66408c992a3216c8). 

Here are the detailed steps I followed:
1) Removed the previous instance of  the ring plugin (sudo npm remove -g homebridge-ring)
2) Stopped the homebridge server (systemctl stop homebridge@1)
3) Installed the alpha plugin as you suggested above (sudo npm install -g homebridge-ring@5.11.0-alpha.0). I got the below two warnings when i installed the plugin:
>npm WARN ws@7.2.1 requires a peer of bufferutil@^4.0.1 but none is installed. You must install peer dependencies yourself.
>npm WARN ws@7.2.1 requires a peer of utf-8-validate@^5.0.2 but none is installed. You must install peer dependencies yourself."
4) Deleted the persist and accessories folder in my homebridge@1 folder (rm -rf /persists, rm -rf /accessories)
5) Removed the "Ring" hub from home app
6) Restarted the homebridge server (systemctl start homebridge@1)
7) Added the accessory again in the home app. After adding, I got a message saying I need to set up the accessories in the manufacturer's app
8) None of the accessories show up in the home app

I also tried to roll back to the public ver 5.11.0 of the plugin and everything seems to work fine. So I think, there might be an issue with the alpha version of the plugin.

Please let me know what logs would help to debug this further. It would also be great if you can let me know how to get rid of the warnings that I get when I try to install the plugin. I never had that earlier.

**Comment by jsrivatsan** [comment]
![7417DF1C-54DB-4CA7-8222-2DF8A55884D6](https://user-images.githubusercontent.com/58554760/70855028-cea43c00-1e78-11ea-97d4-789a6838c19b.png)

**Comment by dgreif** [comment]
@jsrivatsan sorry you ran into so many issue!  Testing the alpha should be possible without jumping through quite so many hoops:

1. `systemctl stop homebridge@1`
2. `sudo npm install -g homebridge-ring@5.11.0-alpha.0`
3. `systemctl start homebridge@1`

The plugin will automatically detect the flood/freeze sensor as a new device and add it to homebridge, so you shouldn't need to delete the `persist` folder or add any new devices in the Home app.  It should just show up once the homebridge server is running again.  I just tried the alpha version in my setup and it works just fine so it must have been some weird fluke after you wiped the persist folder.

**Comment by jsrivatsan** [comment]
@dgreif I tried just updating the plugin as you suggested but I do not see any additional accessories that gets added to the Home app. Here are logs I captured when I restarted homebridge@1. Is there any other setting I need to get new accessories to show up? Thanks a ton for you help and support.

[foo.txt](https://github.com/dgreif/ring/files/3971707/foo.txt)

**Comment by dgreif** [comment]
@jsrivatsan sorry it didn't show up.  My only guess is that they `deviceType` field is something more complicated than `sensor.flood-freeze`.  Could you update to `homebridge-ring@5.11.0-alpha.1` and start check your logs?  I added a new log that should look like `Hidden accessory adapter.ringnet TEST RingNet Adapter`.  If you can find the line in there that relates to your flood/freeze sensor, I should be able to get this ironed out.

**Comment by dgreif** [comment]
I ordered a flood/freeze sensor to test with, should be here tomorrow.  Ironically I left a water valve open and had minor flooding in my kitchen over the weekend 🤦‍♂.  Once I get everything fixed up, I'll do a full release (probably Friday)

**Comment by jsrivatsan** [comment]
@dgreif  Sorry to hear that. Hope the damage was minor. I had water damage as well in my house. So I decided to try out these flood/freeze sensors. 
I have not had much time to try out the alpha1 release. I will try it out over the weekend and give you an update.

**Comment by dgreif** [comment]
@jsrivatsan luckily it was a slow water line that I left open and it only leaked a couple gallons which we were able to dry out.  No damage this time, but the flood sensor will easily pay for itself if I ever make the same mistake!

I got my sensor today and fixed up some mistakes that I made in my previous implementation.  I also switched from a contact sensor to an occupancy sensor to represent the freeze sensor.  Opened vs closed was a little harder to guess at compared to occupied vs unoccupied.  maybe.  Anyways, it's all tested out and is now released in a full release - v`5.11.0`.  Enjoy!
